docs/manual-EN/1.overview/3.design-and-architecture/2.storage-design.md

```diff
@@ -74,69 +73,77 @@ Pict.5 In-key format
+There are two key challenges to implement the Multi Raft Group. **First one is how to share the transport layer**. Because each Raft Group  sends messages to its corresponding peers, if the transport layer cannot be shared, the connection costs will be very high. **Second one is how to design the multi-threading model**. Raft Groups share the same thread pool to prevent staring too many threads and a high context switch cost.
 
 ### Batch
 
-For each Partition, it is necessary to do batch to improve throughput when writing WAL serially. In general, there is nothing special about batch, but **Nebula Graph** designs some special types of WAL based on each part serialization, which brings some challenges.
+For each Partition, it is necessary to do batch multiple operations together to improve throughput when writing WAL serially. In general, there is nothing special about batch, but **Nebula Graph** designs some special types of WAL based on each part serialization, which brings some challenges.
 
-For example, **Nebula Graph** uses WAL to implement lock-free CAS operations, and each CAS operation will execute after all the previous WAL are committed. So for a batch, if there are several WALs in CAS type, we need to divide this batch into several smaller serial groups.
+For example, **Nebula Graph** uses WAL to implement lock-free CAS operations. And every CAS operation will be executed until the previous WAL have beed committed. So for a batch, if there are some logs contain CAS operation, we need to divide this batch into several smaller (sub)groups. And make sure these (sub)groups are executed in sequential order.
 
 ### Learner
 
-New machines need to "catch up" data for quite a long time, and there may be accidents during the process. If a new machine joins the raft group as a follower, it will reduce the HA capacity of the entire cluster. **Nebula Graph** uses the command WAL mentioned above to implement its learner. If the leader encounters the `add learner command` when writing WAL, it will add the learner to its peers and mark it as a learner, so logs will send to them as normal. But the learners will not involve in the leader's election.
+When a new machine is added to a cluster, it has to catch up data for quite a long time. And there may be accidents during this process. If this one directly joins the raft group as a follower role, it will dramatically reduce the availability of the entire cluster. **Nebula Graph** introduce the learner role, and it is implemented by the command WAL mentioned above. When a leader is writing WAL and meets a `add learner command`, it will add the new coming-in learner to its peers list and mark it as a learner. The logs will send to all the peers, both the followers and the learner. But the learner can not vote for the leader's election.
```

Comment:
If this one directly joins the raft group as a follower...
**Nebula Graph** introduces ...
...meets an `add learner command`...
I think `learners` is better than `learners` here